tutorials/cloud-run-with-pomerium-for-end-user-access/index.md

```diff
@@ -0,0 +1,190 @@
+        --role=roles/secretmanager.secretAccessor
+
+    # Deploy Pomerium with policy and configuration references
+    gcloud run deploy pomerium --region us-central1 --platform managed --allow-unauthenticated --max-instances 1 \
+        --image=gcr.io/pomerium-io/pomerium:v0.10.0-rc2-cloudrun \
+        --set-env-vars VALS_FILES="/pomerium/config.yaml:ref+gcpsecrets://${PROJECT}/pomerium-config",POLICY=$(base64 policy.yaml)
+
+    # Set domain mappings for the protected routes and authenticate
+    gcloud beta run domain-mappings --platform managed --region us-central1 create --service=pomerium --domain hello.cloudrun.pomerium.com
+    gcloud beta run domain-mappings --platform managed --region us-central1 create --service=pomerium --domain authn.cloudrun.pomerium.com
+    gcloud beta run domain-mappings --platform managed --region us-central1 create --service=pomerium --domain httpbin.cloudrun.pomerium.com
+
+## Results
+
+### Overview
+
+You should see two applications deployed: The `hello` app is your protected app, and pomerium is... Pomerium!
+
+![Cloud Run overview](https://storage.googleapis.com/gcp-community/tutorials/cloud-run-with-pomerium-for-end-user-access/cloudrun-overview.png)
+
+Pomerium allows unauthenticated access, but `hello` does not.
+
+Here are the domain mappings set up:
+
+![Cloud Run domains](https://storage.googleapis.com/gcp-community/tutorials/cloud-run-with-pomerium-for-end-user-access/cloudrun-domains.png)
+
+### Direct Access
+
+Verify that you can't access the main application directly by visiting
+[`https://hello-direct.cloudrun.pomerium.com`](https://hello-direct.cloudrun.pomerium.com).
+
+![Hello Direct Access](https://storage.googleapis.com/gcp-community/tutorials/cloud-run-with-pomerium-for-end-user-access/hello-direct.png)
+
+You should see a 403 error, because you don't have the proper credentials.
+
+### Authenticated Access
+
+1.  Go to [`https://hello.cloudrun.pomerium.com`](https://hello.cloudrun.pomerium.com).
+
+    You should see a sign-in page:
+    
+    ![Hello sign-in](https://storage.googleapis.com/gcp-community/tutorials/cloud-run-with-pomerium-for-end-user-access/hello-signin.png)
+
+1.  Enter your credentials at the sign-in page.
+
+    After you enter your credentials, you should see a hello page:
+
+    ![Hello](https://storage.googleapis.com/gcp-community/tutorials/cloud-run-with-pomerium-for-end-user-access/hello-success.png)
+
+### Applications not running on Google Cloud
+
+If your target application is not running on Google Cloud, you can also perform your own header validation.
+
+Go to [https://httpbin.cloudrun.pomerium.com](https://httpbin.cloudrun.pomerium.com/headers).
+
+You should see your identity header set:
+
+![Hello](https://storage.googleapis.com/gcp-community/tutorials/cloud-run-with-pomerium-for-end-user-access/headers.png)
+
+See [getting user's identity](https://www.pomerium.com/docs/topics/getting-users-identity.html) for details on using this header.
```

Comment:
maybe at the end, add a `What's Next` section linking to

- pomerium website/config docs
- pomerium's own Cloud Run recipe page.

Reply:
👍